src/compiler/tools/TypeChecker.ts

```diff
@@ -179,6 +180,27 @@ export class TypeChecker {
         return symbol == null ? undefined : this._context.compilerFactory.getSymbol(symbol);
     }
 
+    /**
+     * Gets the resolved signature from a node.
+     * @param node - Node to get the signature from.
+     * @returns undefined if signature cannot be resolved
+     */
+    getResolvedSignature(node: Node<ts.CallLikeExpression>): Signature | undefined {
+        const resolvedSignature = this.compilerObject.getResolvedSignature(node.compilerNode);
+        if (!resolvedSignature || !resolvedSignature.declaration) {
```

Comment:
Why return undefined when `.declaration` doesn't exist? Is that when it's the "unknown signature"? We should probably copy what's done in the compiler API here and not return undefined.

Reply:
let's continue discussion in #550